lib/ansible/utils/display.py

```diff
@@ -18,31 +18,45 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+try:
```

Comment:
So I'm thinking that all of this can go away if we do the remoting stuff I mentioned above and just delegate to the existing `prompt` method, since IIRC it does everything we care about already anyway via `input` (which is totally fine if we synchronize the access to it).